src/bb-modules/Client/Api/Guest.php

```diff
@@ -223,7 +223,7 @@ public function confirm_reset($data)
             throw new \Box_Exception('The link have expired or you have already confirmed password reset.');
         }
 
-        $new_pass = substr(md5(time().uniqid()), 0, 10);
+        $new_pass = substr(hash('sha256', time().random_bytes(13)), 0, 10);
```

Comment:
Doesn't add any any extra security as it is only used to generate a password of 10 characters

I would be more fan sending the user a email with reset link instead or let them pick the password them self 

Sending passwords in plain text is bad...

Reply:
I agree about letting the user choose their own password.

Reply:
I agree as well, but let's focus on that after the initial release. Don't want to keep adding new  things to do before the initial release. As for this specific change, I've actually replaced the hashing function with the password generator function in https://github.com/FOSSBilling/FOSSBilling/pull/385/files